packages/ckeditor5-widget/src/widgettypearound/widgettypearound.js

```diff
@@ -644,6 +647,15 @@ export default class WidgetTypeAround extends Plugin {
 				data.selection = viewDocument.selection;
 			}
 		}, { priority: 'high' } );
+
+		// Note: The priority must precede the default Input plugin compositionstart handler (to call it before delete content).
+		this._listenToIfEnabled( viewDocument, 'compositionstart', () => {
+			if ( model.document.selection.isCollapsed ) {
```

Comment:
What does this check mean?

Reply:
Previously we've done that on `keydown` which is fired just before `compositionstart` on non-Safari browsers. On Safari browsers it's actually fired later, so on master this scenario didn't work at all.

In other words, on Safari we fixed this scenario (it's apparently ok there to modify DOM on `compositionstart`). It didn't work and it works now.

Reply:
I think that the solution for Chrome will be to bring back some handling on `keydown`. But we'll need to be smart to only touch composition-related ones.

Reply:
I reported #12088

Reply:
Actually, I still don't understand that `if()` and it's worth explaining in the code.

Reply:
Bingo, then :D So, to avoid more people being confused about why it's there we need to do something.